Realm/RealmSwift/Object.swift

```diff
@@ -19,40 +19,309 @@
+                   or an `Array` with one object for each persisted property. An exception will be
+                   thrown if any required properties are not present and no default is set.
+
+                   When passing in an `Array`, all properties must be present,
+                   valid and in the same order as the properties defined in the model.
+
+    :returns: The created or updated object.
+
+    :see: Object.primaryKey()
+    */
+    public class func createOrUpdateInDefaultRealmWithObject(object: AnyObject) -> Self {
+        return createOrUpdateInRealm(defaultRealm(), withObject: object)
+    }
+
+    /**
+    Create or update an `Object` in the given `Realm` with the given object.
+
+    This method can only be called on object types with a primary key defined. If there is already
+    an object with the same primary key value in the specified Realm its values are updated and the object
+    is returned. Otherwise this creates and populates a new instance of this object in the specified Realm.
+
+    :param: realm  The Realm in which this object is persisted.
+    :param: object The object used to populate the object. This can be any key/value coding compliant
+                   object, or a JSON object such as those returned from the methods in `NSJSONSerialization`,
+                   or an `Array` with one object for each persisted property. An exception will be
+                   thrown if any required properties are not present and no default is set.
+
+                   When passing in an `Array`, all properties must be present,
+                   valid and in the same order as the properties defined in the model.
+
+    :returns: The created or updated object.
+
+    :see: Object.primaryKey()
+    */
+    public class func createOrUpdateInRealm(realm: Realm, withObject object: AnyObject) -> Self {
+        return unsafeBitCast(RLMCreateObjectInRealmWithValue(realm.rlmRealm, className(), object, .allZeros), self)
+    }
+
+    // MARK: Object Retrieval
+
+    /**
+    Get the single object with the given primary key from the specified Realm,
+    or from the default Realm if the `realm` argument is omitted.
+
+    Returns `nil` if the object does not exist.
+
+    This method requires that `primaryKey()` be overridden on the receiving subclass.
+
+    :see: Object.primaryKey()
+
+    :returns: An object of the subclass type or `nil` if an object with the given primary key does not exist.
+    */
+    public class func objectForPrimaryKey(key: AnyObject, inRealm realm: Realm = defaultRealm()) -> Self? {
+        return unsafeBitCast(RLMGetObject(realm.rlmRealm, className(), key), self)
+    }
+
+    // MARK: Customizing
+
+    /**
+    Return an array of property names for properties which should be indexed. Only supported
+    for string and int properties.
+
+    :returns: `Array` of property names to index.
+    */
+    public class func indexedProperties() -> [String]? { return nil }
```

Comment:
I think that was incompatible with the definition in `RLMObjectBase`, but this isn't the right solution either. I'll comment it out and add a FIXME like the other similar issues in this file.